samza-core/src/main/scala/org/apache/samza/config/JobConfig.scala

```diff
@@ -93,6 +93,7 @@ object JobConfig {
   val JOB_SECURITY_MANAGER_FACTORY = "job.security.manager.factory"
 
   val METADATA_STORE_FACTORY = "metadata.store.factory"
+  val STARTPOINT_METADATA_STORE_FACTORY = "startpoint.metadata.store.factory"
```

Comment:
1. Since the configuration `metadata.store.factory` allows us to have a  metadata store defined per samza job, just wondering if this is really necessary and what value add it provides? 
2. Do we have a use-case where some metadata of the job is stored in one store and the rest of them is stored in a different metadata store?